docs/how-to/hip_porting_driver_api.rst

```diff
@@ -0,0 +1,480 @@
+HIP provides a Context API to facilitate easy porting from existing Driver code.
+In HIP, the ``Ctx`` functions largely provide an alternate syntax for changing the active device.
+
+Most new applications preferentially use ``hipSetDevice`` or the stream APIs. Therefore, HIP has marked the ``hipCtx`` APIs as **deprecated**. Support for these APIs might not be available in future releases. For more details on deprecated APIs, see :doc:`../reference/deprecated_api_list`.
+
+HIP module and Ctx APIs
+=======================
+
+Rather than present two separate APIs, HIP extends the HIP API with new APIs for modules and ``Ctx`` control.
+
+hipModule API
+-------------
+
+Like the CUDA Driver API, the Module API provides additional control over how code is loaded, including options to load code from files or from in-memory pointers.
+NVCC and HIP-Clang target different architectures and use different code object formats. NVCC supports ``cubin`` or ``ptx`` files, while the HIP-Clang path uses the ``hsaco`` format.
+The external compilers which generate these code objects are responsible for generating and loading the correct code object for each platform.
+Notably, there is no fat binary format that can contain code for both NVCC and HIP-Clang platforms. The following table summarizes the formats used on each platform:
+
+.. list-table:: Module formats
+   :header-rows: 1
+
+   * - Format
+     - APIs
+     - NVCC
+     - HIP-CLANG
+   * - Code object
+     - ``hipModuleLoad``, ``hipModuleLoadData``
+     - ``.cubin`` or PTX text
+     - ``.hsaco``
+   * - Fat binary
+     - ``hipModuleLoadFatBin``
+     - ``.fatbin``
+     - ``.hip_fatbin``
+
+``hipcc`` uses HIP-Clang or NVCC to compile host code. Both of these compilers can embed code objects into the final executable. These code objects are automatically loaded when the application starts.
+The ``hipModule`` API can be used to load additional code objects. When used this way, it extends the capability of the automatically loaded code objects.
+HIP-Clang enables both of these capabilities to be used together. Of course, it is possible to create a program with no kernels and no automatic loading.
+
+hipCtx API
+----------
+
+HIP provides a ``Ctx`` API as a thin layer over the existing device functions. The ``Ctx`` API can be used to set the current context or to query properties of the device associated with the context.
+The current context is implicitly used by other APIs, such as ``hipStreamCreate``.
+
+HIPIFY translation of CUDA Driver API
+-------------------------------------
+
+The HIPIFY tools convert CUDA Driver APIs for streams, events, modules, devices, memory management, context, and the profiler to the equivalent HIP calls. For example, ``cuEventCreate`` is translated to ``hipEventCreate``.
+HIPIFY tools also convert error codes from the Driver namespace and coding conventions to the equivalent HIP error code. HIP unifies the APIs for these common functions.
```

Comment:
I'm also not sure if Driver should be capitalized here. If it's the actual name of the namespace in the code, capitalize it.